lib/serviceClient.ts

```diff
@@ -185,9 +199,26 @@ export class ServiceClient {
       options = {};
     }
 
+    if (options.baseUri) {
+      this.baseUri = options.baseUri;
+    }
+
     let serviceClientCredentials: ServiceClientCredentials | undefined;
     if (isTokenCredential(credentials)) {
-      serviceClientCredentials = new AzureIdentityCredentialAdapter(credentials);
+      let scope: string | undefined = undefined;
+      const azureManagementClouds = [
+        "https://management.core.windows.net",
+        "https://management.core.chinacloudapi.cn",
+        "https://management.core.usgovcloudapi.net",
+        "https://management.core.cloudapi.de",
+      ];
+      if (
+        options?.baseUri &&
+        azureManagementClouds.find((cloud) => options!.baseUri!.indexOf(cloud) > -1)
+      ) {
```

Comment:
Can we simplify this?
```suggestion
      if (
        options?.baseUri &&
        azureManagementClouds.includes(options?.baseUri)
      ) {
```

Reply:
Includes breaks in the browser, I think https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

Well, InternetExplorer, so I guess it’s ok to use it

Reply:
I merged it, but keep in mind that includes will work only for exact matches.

Reply:
And that is good right? We want exact matches

Reply:
My brain is kinda fried right now. 🙇‍♂️ 🙇‍♂️ *slowly leaves*

Reply:
As far as we know, the baseuri indeed needs to match the resource manager endpoints we have listed here. If there are cases where this is not true, I would much rather hear that from a customer than make assumptions ourselves

Reply:
Addressed here: https://github.com/Azure/ms-rest-js/pull/457/commits/7224651161fbf7928a08f861a864d5dd34064ac1